internal/rbd/snapshot.go

```diff
@@ -89,6 +89,7 @@ func generateVolFromSnap(rbdSnap *rbdSnapshot) *rbdVolume {
 	vol.RadosNamespace = rbdSnap.RadosNamespace
 	vol.RbdImageName = rbdSnap.RbdSnapName
 	vol.ImageID = rbdSnap.ImageID
+	vol.encryption = rbdSnap.encryption
```

Comment:
You should probably use `rbdSnap.copyEncryptionConfig(vol)` to be safe (and consistent).

Reply:
added now.

Reply:
Just tried this one, as the volumeID remains the same for both snapshot and rbdVolume we are going to get panic.

```text
failed copy encryption config for "replicapool/csi-snap-4cdb2681-a723-11eb-a3f8-82f7194580e5": BUG: "replicapool/csi-vol-3aac6c1c-a723-11eb-a3f8-82f7194580e5" and "replicapool/csi-snap-4cdb2681-a723-11eb-a3f8-82f7194580e5" have the same VolID (0001-0024-dfd8049b-f3c6-4939-bb17-d6dea9d6b281-0000000000000001-4cdb2681-a723-11eb-a3f8-82f7194580e5) set!? Call stack: goroutine 447 [running]:
```

Reply:
Please add a comment that the helper function is not suitable here, or introduce a new helper function. Exceptions in the code need to be documented, or become non-exceptions.